webapp/templates/service-worker.js

```diff
@@ -0,0 +1,43 @@
+// Clean up old caches.
+self.addEventListener(
+  'activate',
+  e => {
+    e.waitUntil(
+      caches.keys()
+        .then(cacheNames => Promise.all(
+          cacheNames
+            .filter(v => '{{ .Version }}' !== v)
+            .map(v => caches.delete(v))
+        ))
+    );
+  }
+);
+
+// Locally cache eligible components/files.
+self.addEventListener(
+  'fetch',
+  e => {
+    return e.respondWith(
+      caches.match(e.request)
+        .then(r => {
+          if (r) {
+            return r;
+          }
+          return fetch(e.request)
+            .then(r => {
+              if (r.ok) {
+                const components = new RegExp('^/((bower_)?components|static)/');
```

Comment:
I suppose with #982 we'll need to cache `node_modules` as well. Shall we do it in this PR?

Reply:
No, I'll do it in #982.